cmd/util/cmd/export-json-execution-state/cmd.go

```diff
@@ -98,6 +102,92 @@ func ExportLedger(ledgerPath string, targetstate string, outputPath string) erro
 			return fmt.Errorf("failed to convert bytes to state: %w", err)
 		}
 	}
+
+	var write func(writer io.Writer) error
+
+	if fullSearch {
+		forest, err := mtrie.NewForest(complete.DefaultCacheSize, noopMetrics, func(evictedTrie *trie.MTrie) error {
+			return nil
+		})
+		if err != nil {
+			return fmt.Errorf("cannot create forest: %w", err)
+		}
+
+		diskWal.PauseRecord()
+
+		sentinel := fmt.Errorf("we_got_the_trie_error")
+
+		rootState := ledger.RootHash(state)
+
+		err = diskWal.ReplayLogsOnly(
+			func(forestSequencing *flattener.FlattenedForest) error {
+				rebuiltTries, err := flattener.RebuildTries(forestSequencing)
+				if err != nil {
+					return fmt.Errorf("rebuilding forest from sequenced nodes failed: %w", err)
+				}
+				err = forest.AddTries(rebuiltTries)
+				if err != nil {
+					return fmt.Errorf("adding rebuilt tries to forest failed: %w", err)
+				}
+
+				for _, trie := range rebuiltTries {
+					rootHash := trie.RootHash()
+					if rootState.Equals(rootHash) {
+						return sentinel
+					}
+				}
```

Comment:
I wonder if rebuilding tries could take in the `rootState` for comparison, so that we could bail as soon as we found a trie that matches the root hash.